src/tools/cephfs/DataScan.cc

```diff
@@ -446,7 +452,28 @@ int DataScan::scan_inodes()
   float progress = 0.0;
   librados::NObjectIterator i = data_io.nobjects_begin(n, m);
 #else
-  librados::NObjectIterator i = data_io.nobjects_begin();
+  librados::NObjectIterator i;
+  bool legacy_filtering = false;
```

Comment:
It looks like you've broken the SHARDED_PGLS build here? legacy_filtering is referenced regardless below, and I imagine it also needs to include the filtering bits. ;)

(Although I'm not sure how interested you are in maintaining the two branches for this short period of time...nor if preprocessor #ifdefs are the right way to handle that, now that I'm seeing them again...)